### DISCUSSION TITLE
How to model (nearly) full-truckload pickup and delivery?

### DISCUSSION THREAD

**Discussion by charlestang06** [reinterpretcat/vrp#152] [Q&A] [answered]
I'm wondering if the following scenario can be modeled with the VRP-CLI package:

Scenario:

- 20-30 vehicles each with capacity 2
- Total of 100+ locations for both delivery and drop off
- a long task list of PU and DO locations with quantities > 2 (meaning multiple vehicles may need to visit a location multiple times)
- Vehicle can continue its route after dropping off load at delivery location (i.e. most of routes will be PU-DO-PU-DO-PU-DO)

I am wondering because pick up/delivery quantities are greater than capacity. If not, are there any potential workarounds for this?

**Comment by reinterpretcat** [comment, marked as answer]
If I got the use case correctly:
- model each PUDO job as multi-job with pickup and delivery job task set (in terms of pragmatic format), see https://reinterpretcat.github.io/vrp/concepts/pragmatic/problem/jobs.html#pickup-and-delivery-job
 - specify location multiple times using multiple jobs if needed
- specify needed number of vehicles with capacity=2, see https://reinterpretcat.github.io/vrp/concepts/pragmatic/problem/vehicles.html#vehicle-types

Currently I'm working on making the project more usable and extendable from code. These changes will be released with the next version.

**Comment by charlestang06** [comment]
What would "specify location multiple times using multiple jobs if needed" mean? Would this just mean I can put as many jobs as I would like in the `plan`?

**Reply by reinterpretcat** [reply]
yes, "plan.jobs" has no limitations on the amount of jobs inside (only hardware limits such as memory and overall speed). Example of usage: https://reinterpretcat.github.io/vrp/examples/pragmatic/basics/job-types.html
